swagger/validate.go

```diff
@@ -0,0 +1,286 @@
+		if isSameURLPath(schemaUrlPath, requestUrlPath) { // TODO: Fix this to identify b/w keyword versus fix words, may be we can give preference to fix words over placeholder values
+			found = true
+			break
+		}
+	}
+
+	// if no exact match is found, checking if any parametric path matches with request URL path
+	if !found {
+		for schemaUrlPath, schemaUrlPathDetail = range swaggerApiSpecPaths.(map[string]interface{}) {
+			if isSameParametricURLPath(schemaUrlPath, requestUrlPath) {
+				found = true
+				break
+			}
+		}
+	}
+
+	// if still no match is found, add a log and return w/o any error
+	if !found {
+		reason := fmt.Sprintf("no json schema is found for schemaUrlPath %s and method %s",
+			schemaUrlPath, requestMethod)
+		fmt.Println(reason)
+		return
+	}
+
+	// as match found, so check if the request method exists
+	methodDetail, found := schemaUrlPathDetail.(map[string]interface{})[strings.ToLower(requestMethod)]
+	if !found {
+		reason := fmt.Sprintf("request method %s is not defined for URL Path: %s",
+			requestMethod, requestUrlPath)
+		fmt.Println(reason)
+		return
+	}
+
+	// if the request method is found, return the request body schema
+	var requestBodyDetail interface{}
+	methodDetailMap := methodDetail.(map[string]interface{})
+	eventNameInterface, found := methodDetailMap["operationId"]
+	if !found {
+		reason := fmt.Sprintf("unable to find event name")
+		fmt.Println(reason)
+		err = errors.New(reason)
+		return
+	}
+	eventName = eventNameInterface.(string)
+
+	requestBodyDetail, found = methodDetailMap["requestBody"]
+	// requestBody will be absent for some APIs (possibly GET and others), and thus, requestBodySchema
+	// will not be defined for such APIs. So, avoid raising error in such case.
+	if requestBodyPresent && !found {
+		reason := fmt.Sprintf("requestBody field is missing in the API schema for URL Path: %s",
+			requestUrlPath)
+		fmt.Println(reason)
+		return
+	}
+
+	if requestBodyPresent { // only if request body is present, do the schema validation
+		requestSchemaMap, err = resolveAndGetRequestBodySchema(requestBodyDetail.(map[string]interface{}), swaggerApiSpecPath)
+	}
+	return
+}
+
+//isSameURLPath checks if the schema's path exactly matches with the request URL path
+func isSameURLPath(schemaURLPath, requestURLPath string) bool {
+	// remove the "/api" from the schemaURKPath before matching with the requestURLPath
+	//schemaURLPath = strings.SplitAfterN(schemaURLPath, "api", 2)[1]
+	return schemaURLPath == requestURLPath
+}
+
+//isSameParametricURLPath checks if the schema's parametric path matches with the request URL path
+func isSameParametricURLPath(schemaURLPath, requestURLPath string) bool {
+	// remove the "/api" from the schemaURLPath before matching with the requestURLPath
+	if !strings.Contains(schemaURLPath, "ops") && !strings.Contains(schemaURLPath, "plugin") &&
+		!strings.Contains(schemaURLPath, "upi") {
```

Comment:
This shud not be a part of a generic library. No Hardcoding shud be done. 
If we want to exclude any internal APIs for validation, we can take a list of Url Pattern as an optional parameter. 

@happay  to confirm

Reply:
Agree with @udsharma015

Reply:
@udsharma015  Okay i'll make the changes.